resources/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,452 @@
+
+### User persona(s)
+<!-- Who is the target user? Include size/org-structure/decision makers where applicable. -->
+
+- Platform engineers: Platform engineers are responsible for streamlining the developer-experience for their organization by defining a set of recommended practices for application teams and provide self-service capabilities for application teams/developers to use. They are responsible for building the platform and providing the necessary tools and services for the application teams to deploy their applications.
+
+- IT Operators: IT operators are responsible for managing the infrastructure and ensuring that the applications are running smoothly. They are responsible for maintaining the infrastructure and providing support for the infrastructure. They are the primary users of Radius Recipes as they are responsible for defining the infrastructure-as-code templates for the applications.
+
+- System Reliability Engineers (SREs) : SREs are responsible for ensuring the reliability of the applications and services that are deployed in the cloud. They are responsible for maintaining the infrastructure and ensuring that the applications are running smoothly. They maintain the infrastructure-as-code templates and provide support for the applications that are deployed. 
+
+- System integrators : System integrators are partners who help enterprises integrate proprietary services with cloud native applications. They are responsible for designing and building cloud native applications on behalf of their customers.
+
+- Developers : Developers are responsible for building the cloud native applications. They are responsible for writing the code, designing and maintaining the applications.
+
+- Open-source contributors : Cloud native open-source contributors can be any of the above personas who are interested in contributing to the cloud native projects. 
+
+### Positive user outcomes
+
+- Author and deploy : I can bring my own applications with custom resources/services and integrate with Radius seamlessly. I don’t have to write Go code to integrate with Radius components and can use simple spec to generate the resource definition and integrate with Radius 
+
+- Recipes for user defined types: I can create Recipes for the custom resource types and deploy via Radius 
+
+- OSS contributions: I can create and contribute Recipes for the custom resource types for the community to use.
```

Comment:
Will the custom resource types be merged into the main repository of Radius, or will we have a separate repository for this?

Reply:
this will be a separate repository with e-2-e validation